test/L2VoteAggregator.t.sol

```diff
@@ -793,3 +831,86 @@ contract Execute is L2VoteAggregatorBase {
     voteAggregator.execute(addrs, exam, te, hi);
   }
 }
+
+contract InternalVotingPeriodEnd is L2VoteAggregatorTest {
+  function testFuzz_CorrectlyCalculateInternalVotingPeriod(
+    uint256 proposalId,
+    uint256 voteStart,
+    uint256 voteEnd,
+    bool isCanceled
+  ) public {
+    voteEnd = bound(voteEnd, voteAggregator.CAST_VOTE_WINDOW(), type(uint256).max);
+    L2GovernorMetadata.Proposal memory proposal = GovernorMetadataMock(
+      address(voteAggregator.GOVERNOR_METADATA())
+    ).createProposal(proposalId, voteStart, voteEnd, isCanceled);
+
+    uint256 lastVotingBlock = voteAggregator.internalVotingPeriodEnd(proposalId);
+    assertEq(lastVotingBlock, proposal.voteEnd - voteAggregator.CAST_VOTE_WINDOW());
+  }
+}
+
+contract ProposalVoteActive is L2VoteAggregatorTest {
+  function testFuzz_ProposalVoteIsActive(uint256 proposalId, uint64 voteStart, uint64 voteEnd)
+    public
+  {
+    voteStart = uint64(bound(voteStart, 0, block.number));
+    voteEnd =
+      uint64(bound(voteEnd, block.number + voteAggregator.CAST_VOTE_WINDOW(), type(uint64).max));
+    GovernorMetadataMock(address(voteAggregator.GOVERNOR_METADATA())).createProposal(
+      proposalId, voteStart, voteEnd, false
+    );
+
+    uint256 lastVotingBlock = voteAggregator.internalVotingPeriodEnd(proposalId);
+
+    vm.roll(lastVotingBlock);
+    bool active = voteAggregator.proposalVoteActive(proposalId);
+    assertEq(active, true, "Proposal is supposed to be active");
+  }
+
+  function testFuzz_ProposalVoteIsInactiveBefore(
+    uint256 proposalId,
+    uint64 voteStart,
+    uint64 voteEnd,
+    bool isCanceled
+  ) public {
+    vm.assume(voteStart > 0); // Prevent underflow because we subtract 1
+    vm.assume(voteStart > block.number); // Block number must be greater than vote start
+    vm.assume(voteEnd > voteAggregator.CAST_VOTE_WINDOW()); //  Prevent underflow
+    vm.assume(voteEnd - voteAggregator.CAST_VOTE_WINDOW() > voteStart); // Proposal must have a
```

Comment:
Instead of using assumes for numerical values like this, it's much more efficient for the test suite to use `bound`, i.e. https://book.getfoundry.sh/reference/forge-std/bound?highlight=bound#bound

When you use assume, it starts the test run then, if the conditions are not met, throws out the execution of this run of the test and starts over with new random values. On the other hand, `bound` takes the random parameter and uses it to derive a value within the range specified. It always returns a value and therefore always executes the test, avoiding a lot of thrashing and overhead by the test runner to find values that meet the requirements. It would be good to update our tests anywhere we're using vm.assume for a numerical value.

Reply:
Yes, definitely we have a separate issue for https://github.com/ScopeLift/l2-flexible-voting/issues/62.

Reply:
As these were moved tests in this pr, I think it makes sense to bundle them into that issue.